frontend/src/components/rideshareEventForm/RideshareEventForm.jsx

```diff
@@ -0,0 +1,368 @@
+import Header from '../header/Header'
+import NearbyEventsMap from '../NearbyEventsMap'
+import axios from 'axios'
+import { useState, useEffect } from 'react'
+import { Menu, MenuButton, MenuItem, MenuItems } from '@headlessui/react'
+import { ChevronDownIcon } from '@heroicons/react/20/solid'
+
+function RideshareEventForm({ user }) {
+    const current_eventId = window.location.pathname.split('/')[3]
+    const [event, setEvent] = useState()
+    const [attendance, setAttendance] = useState()
+    const [hasCar, setHasCar] = useState(false)
+    const numberOfSeats = [1, 2, 3, 4, 5, 6]
+    const [preferedArrivalTime, setPreferedArrivalTime] = useState()
+    const [numberOfSeatsAvailable, setNumberOfSeatsAvailable] = useState(0)
+    const [numberOfSeatsNeeded, setNumberOfSeatsNeeded] = useState(0)
+    const url = import.meta.env.VITE_URL
+    let alreadyRegistered = false
+
+    const mapContainerStyle = {
+        width: '20rem',
+        height: '15rem',
+    }
+
+    useEffect(() => {
+        fetchEvent()
+        fetchAttendance()
+        // To Do:
+        // Fix how to check if a user is registered
+        if (attendance) {
+            if (attendance[0].length > 3) alreadyRegistered = true
+        }
+    }, [])
+
+    const fetchEvent = () => {
+        fetch(`http://localhost:3000/event/${current_eventId}`)
+            .then((response) => response.json())
+            .then((data) => {
+                setEvent(data)
+            })
+            .catch((error) => console.error('Error fetching:', error))
+    }
+
+    const fetchAttendance = () => {
+        fetch(
+            `http://localhost:3000/attendance/user/${user.username}/${current_eventId}`
+        )
+            .then((response) => response.json())
+            .then((data) => {
+                setAttendance(data)
+            })
+            .catch((error) => console.error('Error fetching:', error))
+    }
+
+    const handlePostWithCar = async () => {
+        try {
+            await axios.post(`${url}/ridesharing/withCar`, {
+                seatsAvailable: numberOfSeatsAvailable,
+                arrivalTime: preferedArrivalTime,
+                departingLat: user.lat,
+                departingLong: user.long,
+                user: user,
+                event: event,
+            })
+        } catch (error) {
+            console.error('Error:', error)
+        }
+    }
+
+    const handlePostWithoutCar = async () => {
+        try {
+            await axios.post(`${url}/ridesharing/withoutCar`, {
+                numberOfSeatsNeeded: numberOfSeatsNeeded,
+                preferedArrivalTime: preferedArrivalTime,
+                user: user,
+                event: event,
+            })
+        } catch (error) {
+            console.error('Error:', error)
+        }
+    }
```

Comment:
These two functions are doing the same thing. Just the input is different. Merge them

Reply:
They are posting to different models. Do I merge them?

Reply:
You can merge the two endpoints